common/common.go

```diff
@@ -486,3 +487,21 @@ func GetExitStatus(err error) (int, error) {
 	}
 	return -1, err
 }
+
+// GetArch returns the current architecture. It returns runtime.GOARCH if not on a arm device
+// and armv6l, armv7l, arm64 otherwise
+func GetArch() string {
+	arch := runtime.GOARCH
+	if arch == "arm" {
+		utsName := &syscall.Utsname{}
+		syscall.Uname(utsName)
+		arch = ""
+		for _, b := range utsName.Machine {
+			if b == 0 {
+				break
+			}
+			arch += string(b)
+		}
+	}
+	return arch
```

Comment:
> I don't know if this should be the source of truth

This is a tricky topic as in fact there isn't an well-shared ABI for ARMv7. You may read more at https://wiki.debian.org/ArmHardFloatPort, but in short the FPU part of the ABI is not fixed and is known to have incompatible differences (eg. RaspPi and BeagleBone are both armv7l but binaries are not 100% portable). This resulted in so much [frustation](https://github.com/rust-lang/rust/issues/35590) and bugs, that I mostly agree with golang decision here of defaulting to armv6. For your PR, I think that matching rkt architecture is fine here for the general case, and advanced users may tweak the whole thing (rkt + images) if they need to, provided they know the troubles they are stepping into.

> I didn't even know about the possibility to instruct the linker to do such things. Is this documented somewhere?

It is documented [here](https://golang.org/cmd/compile/). I found it while looking for linker-specific tricks.